{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Populating the interactive namespace from numpy and matplotlib\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/cmb-panasas2/skchoudh/software_frozen/anaconda27/envs/riboraptor/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88\n",
      "  return f(*args, **kwds)\n",
      "/home/cmb-panasas2/skchoudh/software_frozen/anaconda27/envs/riboraptor/lib/python3.5/importlib/_bootstrap.py:222: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88\n",
      "  return f(*args, **kwds)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['hg38', 'mm10', 'sacCerR64', 'MG1655', 'BDGP6']"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%pylab inline\n",
    "\n",
    "import glob\n",
    "import os\n",
    "import pandas as pd\n",
    "from collections import defaultdict\n",
    "from riboraptor.utils import summary_starlogs_over_runs\n",
    "from riboraptor.sradb import SRAdb\n",
    "from riboraptor.helpers import path_leaf, parse_star_logs, millify\n",
    "from riboraptor.cutadapt_to_json import cutadapt_to_json\n",
    "\n",
    "re_ribo_analysis_dir = '/staging/as/skchoudh/re-ribo-analysis/'\n",
    "\n",
    "species = os.listdir(re_ribo_analysis_dir)\n",
    "xl = pd.ExcelFile('../data/datasets/re-ribo-datasets.xlsx')\n",
    "xl.sheet_names  # see all sheet names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "sradb = SRAdb('/staging/as/skchoudh/SRAmetadb.sqlite')\n",
    "geodb = SRAdb('/staging/as/skchoudh/GEOmetadb.sqlite')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "datasets_processed = defaultdict(list)\n",
    "datasets_processed_assemblywise = defaultdict(list)\n",
    "for directory in glob.glob('{}/*/*'.format(re_ribo_analysis_dir)):\n",
    "    srp = path_leaf(directory)\n",
    "    assembly = path_leaf(os.path.dirname(directory))\n",
    "    datasets_processed[srp].append(assembly)\n",
    "    datasets_processed_assemblywise[assembly].append(srp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "datasets = dict([(key, pd.read_excel('../data/datasets/re-ribo-datasets.xlsx', sheet_name=key)) for key in xl.sheet_names])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Publication</th>\n",
       "      <th>Year</th>\n",
       "      <th>SRP</th>\n",
       "      <th>SRP.1</th>\n",
       "      <th>Status</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Adapter provided?</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Guo et al. (2010)</td>\n",
       "      <td>2010</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP002605</td>\n",
       "      <td>Done</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/20703300</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Guo et al. (2010)</td>\n",
       "      <td>2010</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP003554</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/20703300</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Reid and Nicchitta (2012)</td>\n",
       "      <td>2012</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP007963</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22199352</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Stadler &amp; Fire (2011)</td>\n",
       "      <td>2011</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP010374</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22045228</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Hsieh et al. (2012)</td>\n",
       "      <td>2012</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP010679</td>\n",
       "      <td>Done</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22367541</td>\n",
       "      <td>PC3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Fritsch et al. (2012)</td>\n",
       "      <td>2012</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP014542</td>\n",
       "      <td>Done</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22879431</td>\n",
       "      <td>THP-1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Lee et al. (2012)</td>\n",
       "      <td>2012</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP014629</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/22927429</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Stern-Ginossar et al. (2012)</td>\n",
       "      <td>2012</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP016143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/23180859</td>\n",
       "      <td>human foreskin fibroblasts</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Liu et al. (2012)</td>\n",
       "      <td>2012</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP017263</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/23290916</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Loayza-Puch et al. (2013)</td>\n",
       "      <td>2013</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP020544</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/23594524</td>\n",
       "      <td>BJ fibroblast</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Rooijers et al. (2013)</td>\n",
       "      <td>2013</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP027514</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/24301020</td>\n",
       "      <td>BJ fibroblast</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Schueler et al. (2014)</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP028887</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/24417896</td>\n",
       "      <td>MCF7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Stumpf et al. (2013)</td>\n",
       "      <td>2013</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP029589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/24120665</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Gonzalez et al. (2014)</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP031501</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/25122893</td>\n",
       "      <td>Brain</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Guo et al. (2013)</td>\n",
       "      <td>2013</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP031849</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/25070500</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Andreev et al. (2014)</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP038695</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/25621764</td>\n",
       "      <td>HEK293T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Wein et al. (2014)</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP040550</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/25108525</td>\n",
       "      <td>Skeletal muscle</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Menschaert et al. (2015)</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP042937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/25510491</td>\n",
       "      <td>HCT116</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Rutkowski et al.</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP045214</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/25989971</td>\n",
       "      <td>human foreskin fibroblasts</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Gao et al. (2014)</td>\n",
       "      <td>2014</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP049168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>kidney</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Cenik et al. (2015)</td>\n",
       "      <td>2015</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP055009</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26297486</td>\n",
       "      <td>lymphoblastoid</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Ulirsch et al. (2018)</td>\n",
       "      <td>2018</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP092068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/29551269</td>\n",
       "      <td>CD 34</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Blair et al. (2017)</td>\n",
       "      <td>2017</td>\n",
       "      <td>https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...</td>\n",
       "      <td>SRP109126</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/29141229</td>\n",
       "      <td>ESC/NPC</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Publication  Year  \\\n",
       "0              Guo et al. (2010)  2010   \n",
       "1              Guo et al. (2010)  2010   \n",
       "2      Reid and Nicchitta (2012)  2012   \n",
       "3         Stadler & Fire (2011)   2011   \n",
       "4            Hsieh et al. (2012)  2012   \n",
       "5          Fritsch et al. (2012)  2012   \n",
       "6              Lee et al. (2012)  2012   \n",
       "7   Stern-Ginossar et al. (2012)  2012   \n",
       "8              Liu et al. (2012)  2012   \n",
       "9      Loayza-Puch et al. (2013)  2013   \n",
       "10        Rooijers et al. (2013)  2013   \n",
       "11        Schueler et al. (2014)  2014   \n",
       "12          Stumpf et al. (2013)  2013   \n",
       "13        Gonzalez et al. (2014)  2014   \n",
       "14             Guo et al. (2013)  2013   \n",
       "15         Andreev et al. (2014)  2014   \n",
       "16            Wein et al. (2014)  2014   \n",
       "17      Menschaert et al. (2015)  2014   \n",
       "18              Rutkowski et al.  2014   \n",
       "19             Gao et al. (2014)  2014   \n",
       "20           Cenik et al. (2015)  2015   \n",
       "21         Ulirsch et al. (2018)  2018   \n",
       "22           Blair et al. (2017)  2017   \n",
       "\n",
       "                                                  SRP      SRP.1 Status  \\\n",
       "0   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP002605   Done   \n",
       "1   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP003554    NaN   \n",
       "2   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP007963    NaN   \n",
       "3   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP010374    NaN   \n",
       "4   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP010679   Done   \n",
       "5   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP014542   Done   \n",
       "6   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP014629    NaN   \n",
       "7   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP016143    NaN   \n",
       "8   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP017263    NaN   \n",
       "9   https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP020544    NaN   \n",
       "10  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP027514    NaN   \n",
       "11  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP028887    NaN   \n",
       "12  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP029589    NaN   \n",
       "13  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP031501    NaN   \n",
       "14  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP031849    NaN   \n",
       "15  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP038695    NaN   \n",
       "16  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP040550    NaN   \n",
       "17  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP042937    NaN   \n",
       "18  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP045214    NaN   \n",
       "19  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP049168    NaN   \n",
       "20  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP055009    NaN   \n",
       "21  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP092068    NaN   \n",
       "22  https://trace.ncbi.nlm.nih.gov/Traces/sra/?stu...  SRP109126    NaN   \n",
       "\n",
       "                                            PMID                      Tissue  \\\n",
       "0   https://www.ncbi.nlm.nih.gov/pubmed/20703300                        HeLa   \n",
       "1   https://www.ncbi.nlm.nih.gov/pubmed/20703300                        HeLa   \n",
       "2   https://www.ncbi.nlm.nih.gov/pubmed/22199352                      HEK293   \n",
       "3   https://www.ncbi.nlm.nih.gov/pubmed/22045228                        HeLa   \n",
       "4   https://www.ncbi.nlm.nih.gov/pubmed/22367541                         PC3   \n",
       "5   https://www.ncbi.nlm.nih.gov/pubmed/22879431                       THP-1   \n",
       "6   https://www.ncbi.nlm.nih.gov/pubmed/22927429                      HEK293   \n",
       "7   https://www.ncbi.nlm.nih.gov/pubmed/23180859  human foreskin fibroblasts   \n",
       "8   https://www.ncbi.nlm.nih.gov/pubmed/23290916                      HEK293   \n",
       "9   https://www.ncbi.nlm.nih.gov/pubmed/23594524               BJ fibroblast   \n",
       "10  https://www.ncbi.nlm.nih.gov/pubmed/24301020               BJ fibroblast   \n",
       "11  https://www.ncbi.nlm.nih.gov/pubmed/24417896                        MCF7   \n",
       "12  https://www.ncbi.nlm.nih.gov/pubmed/24120665                        HeLa   \n",
       "13  https://www.ncbi.nlm.nih.gov/pubmed/25122893                       Brain   \n",
       "14  https://www.ncbi.nlm.nih.gov/pubmed/25070500                         NaN   \n",
       "15  https://www.ncbi.nlm.nih.gov/pubmed/25621764                     HEK293T   \n",
       "16  https://www.ncbi.nlm.nih.gov/pubmed/25108525             Skeletal muscle   \n",
       "17  https://www.ncbi.nlm.nih.gov/pubmed/25510491                      HCT116   \n",
       "18  https://www.ncbi.nlm.nih.gov/pubmed/25989971  human foreskin fibroblasts   \n",
       "19                                           NaN                      kidney   \n",
       "20  https://www.ncbi.nlm.nih.gov/pubmed/26297486              lymphoblastoid   \n",
       "21  https://www.ncbi.nlm.nih.gov/pubmed/29551269                       CD 34   \n",
       "22  https://www.ncbi.nlm.nih.gov/pubmed/29141229                     ESC/NPC   \n",
       "\n",
       "    Adapter provided?  \n",
       "0                 NaN  \n",
       "1                 NaN  \n",
       "2                 NaN  \n",
       "3                 NaN  \n",
       "4                 NaN  \n",
       "5                 NaN  \n",
       "6                 NaN  \n",
       "7                 NaN  \n",
       "8                 NaN  \n",
       "9                 NaN  \n",
       "10                NaN  \n",
       "11                NaN  \n",
       "12                NaN  \n",
       "13                NaN  \n",
       "14                NaN  \n",
       "15                NaN  \n",
       "16                NaN  \n",
       "17                NaN  \n",
       "18                NaN  \n",
       "19                NaN  \n",
       "20                NaN  \n",
       "21                NaN  \n",
       "22                NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "datasets['hg38']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# How does the mapping rate vary across projects"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "assembly = 'hg38'\n",
    "srp_dfs_modified  = {}\n",
    "\n",
    "for srp in datasets_processed_assemblywise[assembly]:\n",
    "    if os.path.exists(os.path.join(re_ribo_analysis_dir, assembly, srp)):        \n",
    "        srp_df = sradb.sra_convert(srp.split('_')[0])\n",
    "        srp_df.library_layout = srp_df.library_layout.fillna('SINGLE')\n",
    "        srp_df = srp_df[srp_df.library_layout.str.contains('SINGLE')]\n",
    "     \n",
    "        srp_df['pass1_reads_with_adapters'] = None\n",
    "        srp_df['total_reads_raw'] = None\n",
    "        srp_df['pass1_adapter'] = None\n",
    "        srp_df['pass2_adapter'] = None\n",
    "        srp_df['pass2_reads_with_adapters'] = None\n",
    "        srp_df['mapping_total_reads_input'] = None\n",
    "        srp_df['uniquely_mapped'] = None\n",
    "        \n",
    "        \n",
    "        srpdir = os.path.join(re_ribo_analysis_dir, assembly, srp)\n",
    "        starlogsdir = os.path.join(srpdir, 'starlogs')\n",
    "        srp_srx_grouped = srp_df.groupby('experiment_accession')\n",
    "        preprocess_step1_dir = os.path.join(srpdir, 'preprocessed_step1')\n",
    "        preprocess_step2_dir = os.path.join(srpdir, 'preprocessed')\n",
    "        \n",
    "        for srx, srx_group in srp_srx_grouped:\n",
    "            srrs = srx_group['run_accession'].tolist()                             \n",
    "            #starlogs_df = summary_starlogs_over_runs(starlogsdir, srrs)\n",
    "            \n",
    "            for srr in srrs:\n",
    "                starlogs_df = None\n",
    "                if os.path.isfile(os.path.join(starlogsdir, srr+'Log.final.out')):\n",
    "                    starlogs_df = parse_star_logs(os.path.join(starlogsdir, srr+'Log.final.out'))\n",
    "                # Preprocessed_step1 adapter info\n",
    "                step1_txt = os.path.join(preprocess_step1_dir, srr + '.fastq.gz_trimming_report.txt')\n",
    "                step2_txt = os.path.join(preprocess_step2_dir, srr + '_trimmed_trimmed.fq.gz_trimming_report.txt')\n",
    "                step1_cutadapt_json = None\n",
    "                step2_cutadapt_json = None\n",
    "                \n",
    "                if os.path.isfile(step1_txt):\n",
    "                    step1_cutadapt_json = cutadapt_to_json(step1_txt)\n",
    "                \n",
    "                if os.path.isfile(step2_txt):\n",
    "                    step2_cutadapt_json = cutadapt_to_json(step2_txt)\n",
    "                \n",
    "                if step1_cutadapt_json:\n",
    "                    adapters = step1_cutadapt_json['adapters']\n",
    "                    if len(step1_cutadapt_json['adapters']) == 0:\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass1_adapter'] = 'Empty?'                        \n",
    "                    elif isinstance(adapters, str):\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass1_adapter'] = step1_cutadapt_json['adapters']\n",
    "                    else:\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass1_adapter'] = step1_cutadapt_json['adapters']['{} - {}'.format(srr, 'Adapter 1')]\n",
    "                        trim_info1 = step1_cutadapt_json['trim_info'][srr]\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'total_reads_raw'] = trim_info1['r_processed']\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass1_reads_with_adapters'] = trim_info1['r_with_adapters']\n",
    "\n",
    "                if step2_cutadapt_json:                    \n",
    "                    adapters = step2_cutadapt_json['adapters']\n",
    "                    if len(step2_cutadapt_json['adapters']) == 0:\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass2_adapter'] = 'Empty?'   \n",
    "                    elif isinstance(adapters, str):\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass2_adapter'] = step2_cutadapt_json['adapters']\n",
    "                    else:\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass2_adapter'] = step2_cutadapt_json['adapters']['{} - {}'.format(srr + '_trimmed', 'Adapter 1')]                                        \n",
    "                        trim_info2 = step2_cutadapt_json['trim_info'][srr]\n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass2_reads_with_adapters'] = trim_info2['r_with_adapters']               \n",
    "                        srp_df.loc[srp_df.run_accession==srr, 'pass2_total_reads_processed'] = trim_info2['r_processed']\n",
    "                    \n",
    "                if starlogs_df: \n",
    "                    srp_df.loc[srp_df.run_accession==srr, 'mapping_total_reads_input'] = starlogs_df['total_reads']\n",
    "                    srp_df.loc[srp_df.run_accession==srr, 'uniquely_mapped'] = starlogs_df['uniquely_mapped']\n",
    "                    srp_df.loc[srp_df.run_accession==srr, 'uniquely_mapped_percent'] = starlogs_df['uniquely_mapped_percent']\n",
    "                \n",
    "                \n",
    "        cols = ['pass1_reads_with_adapters', 'total_reads_raw', 'uniquely_mapped', 'mapping_total_reads_input']\n",
    "        for col in cols:\n",
    "            srp_df[col] = srp_df[col].apply(lambda z: millify(z))\n",
    "        srp_dfs_modified[srp] = srp_df "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'NaN'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "millify(None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_accession</th>\n",
       "      <th>experiment_accession</th>\n",
       "      <th>run_accession</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_layout</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_name</th>\n",
       "      <th>bases</th>\n",
       "      <th>...</th>\n",
       "      <th>adapter_spec</th>\n",
       "      <th>avg_read_length</th>\n",
       "      <th>pass1_reads_with_adapters</th>\n",
       "      <th>total_reads_raw</th>\n",
       "      <th>pass1_adapter</th>\n",
       "      <th>pass2_adapter</th>\n",
       "      <th>pass2_reads_with_adapters</th>\n",
       "      <th>mapping_total_reads_input</th>\n",
       "      <th>uniquely_mapped</th>\n",
       "      <th>uniquely_mapped_percent</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118285</td>\n",
       "      <td>SRR403882</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869036: mRNA-seq for vehicle treated PC3 ce...</td>\n",
       "      <td>886121960</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>21.6 M</td>\n",
       "      <td>22.2 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>21.5 M</td>\n",
       "      <td>12.7 M</td>\n",
       "      <td>58.91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118286</td>\n",
       "      <td>SRR403883</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869037: Footprint for vehicle treated PC3 c...</td>\n",
       "      <td>830504160</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>20.4 M</td>\n",
       "      <td>20.8 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>20.5 M</td>\n",
       "      <td>3.2 M</td>\n",
       "      <td>15.71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118287</td>\n",
       "      <td>SRR403884</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869038: mRNA-seq for rapamcyin treated PC3 ...</td>\n",
       "      <td>875152240</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>20.9 M</td>\n",
       "      <td>21.9 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>21.1 M</td>\n",
       "      <td>12.6 M</td>\n",
       "      <td>59.63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118288</td>\n",
       "      <td>SRR403885</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869039: Footprint for rapamcyin treated PC3...</td>\n",
       "      <td>789711800</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>19.3 M</td>\n",
       "      <td>19.7 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>18.3 M</td>\n",
       "      <td>3.5 M</td>\n",
       "      <td>19.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118289</td>\n",
       "      <td>SRR403886</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869040: mRNA-seq for PP242 treated PC3 cell...</td>\n",
       "      <td>786234960</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>19.0 M</td>\n",
       "      <td>19.7 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>19.2 M</td>\n",
       "      <td>11.2 M</td>\n",
       "      <td>58.57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118290</td>\n",
       "      <td>SRR403887</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869041: Footprint for PP242 treated PC3 cel...</td>\n",
       "      <td>788273800</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>19.3 M</td>\n",
       "      <td>19.7 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>18.3 M</td>\n",
       "      <td>3.6 M</td>\n",
       "      <td>19.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118291</td>\n",
       "      <td>SRR403888</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869042: mRNA-seq for vehicle treated PC3 ce...</td>\n",
       "      <td>979031600</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>23.7 M</td>\n",
       "      <td>24.5 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>24.3 M</td>\n",
       "      <td>13.6 M</td>\n",
       "      <td>56.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118292</td>\n",
       "      <td>SRR403889</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869043: Footprint for vehicle treated PC3 c...</td>\n",
       "      <td>975917920</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>23.6 M</td>\n",
       "      <td>24.4 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>24.1 M</td>\n",
       "      <td>7.9 M</td>\n",
       "      <td>32.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118293</td>\n",
       "      <td>SRR403890</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869044: mRNA-seq for rapamcyin treated PC3 ...</td>\n",
       "      <td>764698520</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>18.3 M</td>\n",
       "      <td>19.1 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>19.0 M</td>\n",
       "      <td>10.5 M</td>\n",
       "      <td>55.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118294</td>\n",
       "      <td>SRR403891</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869045: Footprint for rapamcyin treated PC3...</td>\n",
       "      <td>1017802720</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>24.8 M</td>\n",
       "      <td>25.4 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>24.1 M</td>\n",
       "      <td>5.3 M</td>\n",
       "      <td>21.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118295</td>\n",
       "      <td>SRR403892</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869046: mRNA-seq for PP242 treated PC3 cell...</td>\n",
       "      <td>1030229480</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>25.2 M</td>\n",
       "      <td>25.8 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>25.6 M</td>\n",
       "      <td>12.9 M</td>\n",
       "      <td>50.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRP010679</td>\n",
       "      <td>SRX118296</td>\n",
       "      <td>SRR403893</td>\n",
       "      <td>9606</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>SINGLE -</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>GSM869047: Footprint for PP242 treated PC3 cel...</td>\n",
       "      <td>1011667000</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>40.0</td>\n",
       "      <td>24.5 M</td>\n",
       "      <td>25.3 M</td>\n",
       "      <td>CTGTAGGCAC</td>\n",
       "      <td>User provided</td>\n",
       "      <td>None</td>\n",
       "      <td>24.8 M</td>\n",
       "      <td>8.7 M</td>\n",
       "      <td>34.94</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   study_accession experiment_accession run_accession  taxon_id  \\\n",
       "0        SRP010679            SRX118285     SRR403882      9606   \n",
       "1        SRP010679            SRX118286     SRR403883      9606   \n",
       "2        SRP010679            SRX118287     SRR403884      9606   \n",
       "3        SRP010679            SRX118288     SRR403885      9606   \n",
       "4        SRP010679            SRX118289     SRR403886      9606   \n",
       "5        SRP010679            SRX118290     SRR403887      9606   \n",
       "6        SRP010679            SRX118291     SRR403888      9606   \n",
       "7        SRP010679            SRX118292     SRR403889      9606   \n",
       "8        SRP010679            SRX118293     SRR403890      9606   \n",
       "9        SRP010679            SRX118294     SRR403891      9606   \n",
       "10       SRP010679            SRX118295     SRR403892      9606   \n",
       "11       SRP010679            SRX118296     SRR403893      9606   \n",
       "\n",
       "   library_selection library_layout library_strategy  library_source  \\\n",
       "0               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "1               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "2               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "3               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "4               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "5               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "6               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "7               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "8               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "9               cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "10              cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "11              cDNA      SINGLE -           RNA-Seq  TRANSCRIPTOMIC   \n",
       "\n",
       "                                         library_name       bases  \\\n",
       "0   GSM869036: mRNA-seq for vehicle treated PC3 ce...   886121960   \n",
       "1   GSM869037: Footprint for vehicle treated PC3 c...   830504160   \n",
       "2   GSM869038: mRNA-seq for rapamcyin treated PC3 ...   875152240   \n",
       "3   GSM869039: Footprint for rapamcyin treated PC3...   789711800   \n",
       "4   GSM869040: mRNA-seq for PP242 treated PC3 cell...   786234960   \n",
       "5   GSM869041: Footprint for PP242 treated PC3 cel...   788273800   \n",
       "6   GSM869042: mRNA-seq for vehicle treated PC3 ce...   979031600   \n",
       "7   GSM869043: Footprint for vehicle treated PC3 c...   975917920   \n",
       "8   GSM869044: mRNA-seq for rapamcyin treated PC3 ...   764698520   \n",
       "9   GSM869045: Footprint for rapamcyin treated PC3...  1017802720   \n",
       "10  GSM869046: mRNA-seq for PP242 treated PC3 cell...  1030229480   \n",
       "11  GSM869047: Footprint for PP242 treated PC3 cel...  1011667000   \n",
       "\n",
       "             ...            adapter_spec avg_read_length  \\\n",
       "0            ...                    None            40.0   \n",
       "1            ...                    None            40.0   \n",
       "2            ...                    None            40.0   \n",
       "3            ...                    None            40.0   \n",
       "4            ...                    None            40.0   \n",
       "5            ...                    None            40.0   \n",
       "6            ...                    None            40.0   \n",
       "7            ...                    None            40.0   \n",
       "8            ...                    None            40.0   \n",
       "9            ...                    None            40.0   \n",
       "10           ...                    None            40.0   \n",
       "11           ...                    None            40.0   \n",
       "\n",
       "    pass1_reads_with_adapters total_reads_raw pass1_adapter  pass2_adapter  \\\n",
       "0                      21.6 M          22.2 M    CTGTAGGCAC  User provided   \n",
       "1                      20.4 M          20.8 M    CTGTAGGCAC  User provided   \n",
       "2                      20.9 M          21.9 M    CTGTAGGCAC  User provided   \n",
       "3                      19.3 M          19.7 M    CTGTAGGCAC  User provided   \n",
       "4                      19.0 M          19.7 M    CTGTAGGCAC  User provided   \n",
       "5                      19.3 M          19.7 M    CTGTAGGCAC  User provided   \n",
       "6                      23.7 M          24.5 M    CTGTAGGCAC  User provided   \n",
       "7                      23.6 M          24.4 M    CTGTAGGCAC  User provided   \n",
       "8                      18.3 M          19.1 M    CTGTAGGCAC  User provided   \n",
       "9                      24.8 M          25.4 M    CTGTAGGCAC  User provided   \n",
       "10                     25.2 M          25.8 M    CTGTAGGCAC  User provided   \n",
       "11                     24.5 M          25.3 M    CTGTAGGCAC  User provided   \n",
       "\n",
       "   pass2_reads_with_adapters mapping_total_reads_input uniquely_mapped  \\\n",
       "0                       None                    21.5 M          12.7 M   \n",
       "1                       None                    20.5 M           3.2 M   \n",
       "2                       None                    21.1 M          12.6 M   \n",
       "3                       None                    18.3 M           3.5 M   \n",
       "4                       None                    19.2 M          11.2 M   \n",
       "5                       None                    18.3 M           3.6 M   \n",
       "6                       None                    24.3 M          13.6 M   \n",
       "7                       None                    24.1 M           7.9 M   \n",
       "8                       None                    19.0 M          10.5 M   \n",
       "9                       None                    24.1 M           5.3 M   \n",
       "10                      None                    25.6 M          12.9 M   \n",
       "11                      None                    24.8 M           8.7 M   \n",
       "\n",
       "   uniquely_mapped_percent  \n",
       "0                    58.91  \n",
       "1                    15.71  \n",
       "2                    59.63  \n",
       "3                    19.17  \n",
       "4                    58.57  \n",
       "5                    19.48  \n",
       "6                    56.04  \n",
       "7                    32.98  \n",
       "8                    55.27  \n",
       "9                    21.96  \n",
       "10                   50.28  \n",
       "11                   34.94  \n",
       "\n",
       "[12 rows x 21 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "srp_dfs_modified['SRP010679_fixed_adapters']#.pass1_adapter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['study_accession', 'experiment_accession', 'run_accession', 'taxon_id',\n",
       "       'library_selection', 'library_layout', 'library_strategy',\n",
       "       'library_source', 'library_name', 'bases', 'spots', 'adapter_spec',\n",
       "       'avg_read_length', 'pass1_adapter', 'total_reads_raw',\n",
       "       'pass1_reads_with_adapters', 'pass2_adapter',\n",
       "       'mapping_total_reads_input', 'uniquely_mapped',\n",
       "       'uniquely_mapped_percent'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     22.2 M\n",
       "1     20.8 M\n",
       "2     21.9 M\n",
       "3     19.7 M\n",
       "4     19.7 M\n",
       "5     19.7 M\n",
       "6     24.5 M\n",
       "7     24.4 M\n",
       "8     19.1 M\n",
       "9     25.4 M\n",
       "10    25.8 M\n",
       "11    25.3 M\n",
       "Name: total_reads_raw, dtype: object"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "x['total_reads_raw'].apply(lambda z: millify(z))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['SRP038695', 'SRP098789', 'SRP017942', 'SRP042937', 'SRP049168', 'SRP055009', 'SRP109126', 'SRP092068', 'SRP010679_fixed_adapters', 'SRP045214'])"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "srp_dfs_modified.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'adapters': 'User provided',\n",
       " 'length_counts': {},\n",
       " 'length_exp': {},\n",
       " 'length_obsexp': {},\n",
       " 'trim_info': {}}"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "step2_cutadapt_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'multi_mapped': 32008381,\n",
       " 'multi_mapped_percent': 78.38,\n",
       " 'total_reads': 40839670,\n",
       " 'uniquely_mapped': 6889788,\n",
       " 'uniquely_mapped_percent': 16.87,\n",
       " 'unmapped_percent': 4.75}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "starlog_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/staging/as/skchoudh/re-ribo-analysis/hg38/SRP010679_fixed_adapters/starlogs'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "starlogsdir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'adapters': {'SRR403893 - Adapter 1': 'CTGTAGGCAC'},\n",
       " 'length_counts': {'SRR403893 - Adapter 1': {1: 372801,\n",
       "   2: 191462,\n",
       "   3: 253301,\n",
       "   4: 401023,\n",
       "   5: 814034,\n",
       "   6: 1548132,\n",
       "   7: 2984751,\n",
       "   8: 3972228,\n",
       "   9: 4716147,\n",
       "   10: 4178020,\n",
       "   11: 2376079,\n",
       "   12: 1130101,\n",
       "   13: 427355,\n",
       "   14: 217874,\n",
       "   15: 135002,\n",
       "   16: 87970,\n",
       "   17: 77490,\n",
       "   18: 72771,\n",
       "   19: 34984,\n",
       "   20: 19913,\n",
       "   21: 12647,\n",
       "   22: 15391,\n",
       "   23: 45581,\n",
       "   24: 7629,\n",
       "   25: 11707,\n",
       "   26: 7654,\n",
       "   27: 7280,\n",
       "   28: 12423,\n",
       "   29: 14146,\n",
       "   30: 13855,\n",
       "   31: 13116,\n",
       "   32: 4182,\n",
       "   33: 1605,\n",
       "   34: 1156,\n",
       "   35: 1575,\n",
       "   36: 1380,\n",
       "   37: 1685,\n",
       "   38: 24914,\n",
       "   39: 31596,\n",
       "   40: 288682}},\n",
       " 'length_exp': {'SRR403893 - Adapter 1': {1: 6322918.8,\n",
       "   2: 1580729.7,\n",
       "   3: 395182.4,\n",
       "   4: 98795.6,\n",
       "   5: 24698.9,\n",
       "   6: 6174.7,\n",
       "   7: 1543.7,\n",
       "   8: 385.9,\n",
       "   9: 96.5,\n",
       "   10: 24.1,\n",
       "   11: 24.1,\n",
       "   12: 24.1,\n",
       "   13: 24.1,\n",
       "   14: 24.1,\n",
       "   15: 24.1,\n",
       "   16: 24.1,\n",
       "   17: 24.1,\n",
       "   18: 24.1,\n",
       "   19: 24.1,\n",
       "   20: 24.1,\n",
       "   21: 24.1,\n",
       "   22: 24.1,\n",
       "   23: 24.1,\n",
       "   24: 24.1,\n",
       "   25: 24.1,\n",
       "   26: 24.1,\n",
       "   27: 24.1,\n",
       "   28: 24.1,\n",
       "   29: 24.1,\n",
       "   30: 24.1,\n",
       "   31: 24.1,\n",
       "   32: 24.1,\n",
       "   33: 24.1,\n",
       "   34: 24.1,\n",
       "   35: 24.1,\n",
       "   36: 24.1,\n",
       "   37: 24.1,\n",
       "   38: 24.1,\n",
       "   39: 24.1,\n",
       "   40: 24.1}},\n",
       " 'length_obsexp': {'SRR403893 - Adapter 1': {1: 0.05896027005755633,\n",
       "   2: 0.12112254232965952,\n",
       "   3: 0.6409723712392049,\n",
       "   4: 4.059118017401585,\n",
       "   5: 32.95830988424586,\n",
       "   6: 250.72181644452363,\n",
       "   7: 1933.5045669495369,\n",
       "   8: 10293.412801243847,\n",
       "   9: 48871.9896373057,\n",
       "   10: 173361.82572614108,\n",
       "   11: 98592.48962655601,\n",
       "   12: 46892.15767634854,\n",
       "   13: 17732.57261410788,\n",
       "   14: 9040.414937759335,\n",
       "   15: 5601.7427385892115,\n",
       "   16: 3650.207468879668,\n",
       "   17: 3215.3526970954354,\n",
       "   18: 3019.54356846473,\n",
       "   19: 1451.6182572614107,\n",
       "   20: 826.2655601659751,\n",
       "   21: 524.7717842323651,\n",
       "   22: 638.6307053941908,\n",
       "   23: 1891.3278008298755,\n",
       "   24: 316.55601659751034,\n",
       "   25: 485.76763485477176,\n",
       "   26: 317.5933609958506,\n",
       "   27: 302.0746887966805,\n",
       "   28: 515.4771784232365,\n",
       "   29: 586.9709543568464,\n",
       "   30: 574.896265560166,\n",
       "   31: 544.2323651452282,\n",
       "   32: 173.52697095435684,\n",
       "   33: 66.59751037344398,\n",
       "   34: 47.96680497925311,\n",
       "   35: 65.35269709543569,\n",
       "   36: 57.26141078838174,\n",
       "   37: 69.91701244813278,\n",
       "   38: 1033.7759336099584,\n",
       "   39: 1311.03734439834,\n",
       "   40: 11978.50622406639}},\n",
       " 'trim_info': {'SRR403893': {'bp_processed': 1011667000,\n",
       "   'bp_written': 774708359,\n",
       "   'quality_trimmed': 9932999,\n",
       "   'r_processed': 25291675,\n",
       "   'r_with_adapters': 24529642}}}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "step1_cutadapt_json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "srp_df['total_reads'] =  None\n",
    "        srp_df['total_reads_post_trimming'] = Nonesrp_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "files_not_found"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# How has the adapter evolved over the years?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:riboraptor]",
   "language": "python",
   "name": "conda-env-riboraptor-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
